### ISSUE TITLE
Json serialization for map throwing error

### ISSUE THREAD

**Issue by spino17** [JairusSW/as-json#80]
I am trying to compile and run below code:
```typescript

import { JSON } from "json-as/assembly";

@json
class Inner {
  name: string;

  constructor(name: string) {
    this.name = name;
  }
}

@json
class Outer {
  name: string;
  m: Map<string, Inner>;

  constructor(name: string) {
    this.name = name;
    this.m = new Map();
  }
}

export function init(): void {
  let serializedState = JSON.stringify<Outer>(new Outer("bhavya"));
}
```

command: `npm run asbuild` runs fine. When I am running it using the `release.js` wrapped function, it is throwing below error:
```
wasm://wasm/0001a6ba:1


RuntimeError: memory access out of bounds
    at wasm://wasm/0001a6ba:wasm-function[25]:0x159f
    at wasm://wasm/0001a6ba:wasm-function[26]:0x5905
    at wasm://wasm/0001a6ba:wasm-function[27]:0x5ce2
    at file:///<BASE-FILE-PATH>/tests/index.js:2:13
```
. And if I replace `Inner` with `u64` like below:
```typescript

@json
class Outer {
  name: string;
  m: Map<string, u64>;

  constructor(name: string) {
    this.name = name;
    this.m = new Map();
  }
}
```
It is working fine.

**Comment by JairusSW** [comment]
Thanks, taking a look in a moment

**Comment by spino17** [comment]
Thanks @JairusSW for the response. Let me know if you need more information about the system on which I am running the example.

**Comment by JairusSW** [comment]
Just checking, but your using `v0.9.8` (latest) I presume?

**Comment by spino17** [comment]
No, I am using ^0.8.8 and visitor-as with ^0.11.4

**Comment by JairusSW** [comment]
Looks like I didn't support serializing empty maps (`{}`). Fixed in latest version `v0.9.9`

**Comment by spino17** [comment]
Thanks @JairusSW for mentioning this! It's working fine with this version.
